ow_power_system/src/power_system_node.cpp

```diff
@@ -29,22 +29,41 @@ int main(int argc, char* argv[]) {
   ros::Publisher RUL_pub = n.advertise<std_msgs::Float64>("remaining_useful_life",1000);
   
   //Load power values csv
-  string csv_path,csv_file;
-  csv_file = "/data/onewatt.csv";
-  //FILE OPTIONS: onewatt.csv, eightwatt.csv, sixteenwatt.csv
-  csv_path = ros::package::getPath("ow_power_system");
-  csv_path += csv_file;
+  string csv_path;
+  auto csv_path_param_exist = n.param("power_draw_csv_path", csv_path,
+    ros::package::getPath("ow_power_system") + "/data/onewatt.csv");
+
+  if (!csv_path_param_exist)
+  {
+    ROS_WARN_NAMED("power_system_node", "power_draw_csv_path param was not set! Using default value: onewatt.csv");
+  }
 
   auto power_csv = loadCSV(csv_path);
-  
-  //Define our publication rate and initialize our time to 0
-  double power_update_rate;  
-  if (csv_file=="/data/onewatt.csv"){
-    power_update_rate = 0.10; //csv is at 10Hz intervals
-  } else {
-    power_update_rate = 1.0; // 1 Hz default
+
+  // Retrieve our publication rate expressed in Hz
+  double power_update_rate;
+  auto update_rate_param_exist = n.param("power_update_rate", power_update_rate, 10.0);
+
+  if (!update_rate_param_exist)
+  {
+    ROS_WARN_NAMED("power_system_node", "power_update_rate param was not set! Using default value: 10 Hz");
   }
-  int t_step= 0;
+  else
+  {
+    // Validated the parameter
+    if (power_update_rate == 0)
+    {
+      power_update_rate = 10;
+      ROS_WARN_NAMED("power_system_node", "power_update_rate param was set to zero! Using default value: 10 Hz");
+    }
```

Comment:
Maybe this should be an error, I am not sure.
Or instead we can consider that setting a value of zero means using the highest rate possible (by convention) .

Reply:
It looks like Rate uses the exact frequency, right? http://docs.ros.org/diamondback/api/rostime/html/rate_8cpp_source.html
We want onewatt.csv to update every 10 seconds and the others to update every 1 second.